task4.cpp

```diff
@@ -0,0 +1,34 @@
+#include <iostream>
+using namespace std;
+struct Students {
+    char name[30];
+    unsigned int facNum;
+};
+void readStudent(Students& student, char(&name)[30], int& facNum,int& index) {
+    index = 0;
+    cout << "Enter student name:" << endl;
+    cin.getline(name, 30);
+    cout << "Enter faculty numbeer:" << endl;
+    cin >> facNum;
+    while (name[index] != '\0') {
+        student.name[index] = name[index];
+        index++;
+    }
+    student.facNum = facNum;
+}
+void printStudent(Students& student,int& index) {
+    cout << "Name:";
+    for(int j=0;j<index;j++) {
+        cout << student.name[j];
+    }
+    cout << endl;
+    cout << "Faculty number:" << student.facNum;
+}
+int main()
+{
+    char name[30];
+    int facNum,index;
+    Students student;
+    readStudent(student, name, facNum,index);
+    printStudent(student,index);
```

Comment:
Also you should try these operations in the main
Student st = { "Kolio", 10 };
Student *pSt = &st;
const Student* pCSt = &st;
Student* const cPSt = &st;
const Student* const cpCSt = &st;
Student& refSt = st;
const Student& refCSt = st;